e2e/test/scenarios/collections/collection-pinned-overview.cy.spec.js

```diff
@@ -199,6 +199,28 @@ describe("scenarios > collection pinned items overview", () => {
       cy.findByText("A question").should("be.visible");
     });
   });
+
+  it("should be able to pin a visualization by dragging it up", () => {
+    cy.request("PUT", "/api/card/2", {
+      collection_position: 1,
+      collection_preview: false,
+    });
+    openRootCollection();
+
+    cy.findByTestId("collection-table")
+      .findByText("Orders, Count, Grouped by Created At (year)")
+      .as("draggingViz");
+
+    cy.findByTestId("pinned-items").as("pinnedItems");
+
+    // this test can give us some degree of confidence, but its effectiveness is limited
+    // because we are manually firing events on the correct elements. It doesn't seem that there's
+    // a way to actually simulate the raw user interaction of dragging a certain distance in cypress.
+    // this will not guarantee that the drag and drop functionality will work in the real world, e.g
+    // when our various drag + drop libraries start interfering with events on one another.
+    // for example, this test would not have caught https://github.com/metabase/metabase/issues/30614
+    dragAndDrop("draggingViz", "pinnedItems");
```

Comment:
@nemanjaglumac I wonder if you know if there's any way to test drag + drop that doesn't involve specifying which events to fire on which elements? that actually doesn't help catch bugs that are caused by events not getting fired on the intended elements when they're dragged over.

Reply:
I took a close look at both https://github.com/dmtrKovalenko/cypress-real-events and https://github.com/4teamwork/cypress-drag-drop and neither allow you to do what we really need here - simulate a mouse moving over a screen region - they both rely on firing events on specific elements - the bugfix here is that the events got fired on the wrong elements while hovering the right screen region - so any test where we tell it where the fire the events will not fail. 😢